### PR TITLE
Implement Actions extractor and placeholder Actions QL packs

### PR DESCRIPTION
**Pull request by dbartol**
The PR implements an extractor for GitHub Actions workflows. The extractor more or less just forwards to the JavaScript extractor to do the actual extraction. The JavaScript extractor handles both JS and YAML. For now, we're only using the YAML, but we have ambitions to analyze JS code embedded within a workflow in the future. The `autobuild.*` scripts copy all of the `CODEQL_EXTRACTOR_ACTIONS_*` environment variables into the `CODEQL_EXTRACTOR_JAVASCRIPT_*` variables for the JavaScript extractor. In addition, if no path filters are specified by the user, we supply a default set of path filters the extractor only YAML files that are likely to be Actions workflows or reusable Actions.

In addition to the extractor itself, I've added placeholder CodeQL packs for the library, queries, and tests. There are two placeholder tests: one that imports a `.qll` file from the library pack, and one that runs a query from the query pack. Note that the library pack does _not_ have a dbscheme. Instead, it depends on `codeql/javascript-all`, since the actual extraction is done by the JavaScript extractor with the JavaScript dbscheme. The `extractor: actions` property on the query pack ensures that the queries are treated as being for the `actions` language, rather than the `javascript` language.

Note to reviewers: I know nothing about Bash except what I've Googled in the last few hours. Please take a close look at the `autobuild.sh` script.

A subsequent PR will add Bazel files to incorporate the new extractor into the CLI build.

### CODE REVIEW

**Comment by dbartol** [comment]
@pwntester

**Comment by RasmusWL** [comment]
Since I had the code checked out to fix  the formatting, I took the liberty of adding auto-labeler support as well :blush:

File: actions/extractor/tools/autobuild.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+DEFAULT_PATH_FILTERS=$(cat << END
+exclude:**/*
+include:.github/workflows/**/*.yml
+include:.github/workflows/**/*.yaml
+include:**/action.yml
+include:**/action.yaml
+END
+)
+
+if [ -n "${LGTM_INDEX_INCLUDE:-}" ] || [ -n "${LGTM_INDEX_EXCLUDE:-}" ] || [ -n "${LGTM_INDEX_FILTERS:-}" ] ; then
```
Comment (RasmusWL) [inline actions/extractor/tools/autobuild.sh:14]: Minor nitpick: no reason to add default value with empty string if we're just testing if they're not null anyway :shrug: 

```suggestion
if [ -n "${LGTM_INDEX_INCLUDE}" ] || [ -n "${LGTM_INDEX_EXCLUDE}" ] || [ -n "${LGTM_INDEX_FILTERS}" ] ; then
```

**Review by RasmusWL — APPROVED** [review]
This looks plausible to me :+1: I did a quick grep of uses of `CODEQL_EXTRACTOR_JAVASCRIPT_*` env vars and I didn't spot anything missing.

Bash parts LGTM :+1: (just a minor nitpick)

**Review by aibaars — CHANGES_REQUESTED** [review]
I think the change to be bash script may cause problems if the environment variables are uninitialised. Did you test the change?

**Comment by dbartol** [comment]
> I think the change to be bash script may cause problems if the environment variables are uninitialised. Did you test the change?

I'll give them a try locally. I'm planning integration tests once everything is hooked up into the CLI build.
